inc/device/class/accelerometer.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "device/base.hpp"
+
+namespace exo
+{
+namespace device
+{
+namespace cls
+{
+
+/**
+ * @brief      Accelerometer is to be used as a generic interface to be used
+ *             as the base for implementations of specific accelerometers.
+ *
+ * @tparam     ACCEL_TYPE  Numerical storage type to be used for representing
+ *                         acceleration values.
+ * @tparam     DOF         Number of 'degrees-of-freedom'
+ * @tparam     SPATIAL_UNIT Unit of measure used to represent the sensor's location within the robot's reference frame
+ */
+template <class ACCEL_TYPE, size_t DOF, class SPATIAL_UNIT=exo::units::M>
+struct Accelerometer : public exo::device::Base<SPATIAL_UNIT>
```

Comment:
IMU might be a better term to use here, since acceleration is really only one of the many things you would likely be getting from it. Most IMU's are at least 9 DOF, where only a third is the acceleration (the other being angular orientation and magnetic field).

Reply:
Ah. when I scroll down further I see you are breaking each out individually. That's cool with me. Why support setting the number of DoF then? I don't see us ever using a 1 or 2d accelerometer for anything.

Reply:
Agreed. We definitely won't be, but exo is intended to be flexible, so I was just trying to think of the future 🙂